rq/worker.py

```diff
@@ -262,6 +263,26 @@ def __init__(self, queues, name: t.Optional[str] = None, default_result_ttl=DEFA
         elif exception_handlers is not None:
             self.push_exc_handler(exception_handlers)
 
+    def _set_connection(self, connection: Optional['Redis'], default_worker_ttl: int) -> 'Redis':
+        """Improves the reliability of the Worker's connection to the Queue.
+        Will set an ExponentialBackoff logic to the connection error's, and 
+        set a `socket_timout` in case the `BLPOP` command hangs (or any other command).
+
+        Args:
+            connection (Optional[Redis]): The Redis Connection.
+            default_worker_ttl (int): The Default Worker TTL
+        """
+        if connection is None:
+            connection = get_current_connection()    
+        retry = RedisRetry(RedisBackoff(), 3)
+        timeout = default_worker_ttl - 5
```

Comment:
Take a look at how the actual timeout is calculated here. We need to base `socket_timeout` based on this. https://github.com/rq/rq/blob/fa7a635aa42b9626abf488d51863b2333d7837e5/rq/worker.py#L623